### PR TITLE
BUG: Fix some index_col tests for pyarrow read_csv

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers/arrow_parser_wrapper.py [hunk 1]
```diff
@@ -91,20 +91,35 @@ def _finalize_output(self, frame: DataFrame) -> DataFrame:
...
         # we only need the frame not the names
         frame.columns, frame = self._do_date_conversions(frame.columns, frame)
         if self.index_col is not None:
             for i, item in enumerate(self.index_col):
                 if is_integer(item):
                     self.index_col[i] = frame.columns[item]
+                else:
+                    # String case
+                    if item not in frame.columns:
+                        raise ValueError(f"Index {item} invalid")
```
Comment (jreback) [inline pandas/io/parsers/arrow_parser_wrapper.py:118]: hit in tests? is this an error which is common across parsers? should this be a KeyError? (this is basically an invalid index_cocl right?

**Review by jreback — CHANGES_REQUESTED** [review]
cc @gfyoung

**Comment by jreback** [comment]
add this PR number on the same release note for the arrow reader

File: pandas/io/parsers/arrow_parser_wrapper.py (hunk 1 above)
Comment (lithomas1) [reply pandas/io/parsers/arrow_parser_wrapper.py:118]: Yup, idk why it's a ValueError, though. It's hit by tests though.
https://github.com/pandas-dev/pandas/pull/43706/files#diff-6bbd84495202b9fe8941c1bdfaa4ac1077735e8334abf809b2830827fdd6dd4eR42-R45

**Comment by lithomas1** [comment]
> add this PR number on the same release note for the arrow reader

This shouldn't be necessary, as its unreleased. I'm planning a lot more changes to the Arrow CSV reader to bring up somewhat up to par, and this would add clutter to the release notes.

**Comment by jreback** [comment]
> > add this PR number on the same release note for the arrow reader
> 
> This shouldn't be necessary, as its unreleased. I'm planning a lot more changes to the Arrow CSV reader to bring up somewhat up to par, and this would add clutter to the release notes.

it doesn't matter. the arrow change is now composed of multiple PRs which is totally fine (an alternative if you want is to link these follow PRs to the original at the top). The point is in theory we can point to a single issue / PR to have the full change.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @lithomas1

**Pull request merged**